### DISCUSSION TITLE
Updating arbitrary Dockerfile environment variables or arguments with Renovate

### DISCUSSION THREAD

**Discussion by fdorssers** [renovatebot/renovate#32510] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

Hi all,

I've got various environment variables / arguments in `Dockerfile` that I would like renovate to update, but I can't really figure out _how_. 

For example I've got this in a dockerfile:
```python
# renovate: datasource=pypi depName=uv
ARG UV_VERSION=0.4.24
```

Based on examples [here](https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile#L1) and [here](https://github.com/renovatebot/base-image/blob/main/Dockerfile#L77) I was expecting renovate to recognize that it should check `uv` on `pypi`. However that doesn't happen. 

I've also found [this page](https://docs.renovatebot.com/modules/manager/regex/) that talks about adding custom managers with regex (and which is also referred to [here](https://github.com/renovatebot/renovate/issues/839)), but when I check the other example repos that I linked before I can't seem to find those definitions. So I was assuming that this was supported by default.

I feel like I'm missing something relatively straightforward. Does anyone have an idea? Is it really just a matter of implementing a custom regex manager? I've got a minimal reproduction available [here](https://github.com/fdorssers/renovate-docker-arg-discussion-32510).

**Comment by rarkins** [comment, marked as answer]
You'll need to write your own custom regex manager OR reuse some of the ones in this file: https://docs.renovatebot.com/presets-customManagers/#custommanagersdockerfileversions

**Reply by fdorssers** [reply]
Aaah awesome, thanks. Now that I followed it a bit further I also found where it was defined for the renovatebot base-image ([in the `.github` repo](https://github.com/renovatebot/.github/blob/e9b4b2468173add62089369e306e181343d31caf/default.json#L165)). 
I initially wasn't able to find the connection, so I was wrongly assuming that it was some default functionality. Thanks for pointing out that example!
